solidity/test/rewards/TestRewardUpgrades.js

```diff
@@ -0,0 +1,222 @@
+const { accounts, contract, web3 } = require("@openzeppelin/test-environment")
+const { createSnapshot, restoreSnapshot } = require("../helpers/snapshot.js")
+const { expectRevert, time } = require("@openzeppelin/test-helpers")
+
+const KeepToken = contract.fromArtifact('KeepToken')
+const RewardsStub = contract.fromArtifact("RewardsStub");
+const NewRewardsStub = contract.fromArtifact("NewRewardsStub");
+
+const BN = web3.utils.BN
+const chai = require('chai')
+chai.use(require('bn-chai')(BN))
+const expect = chai.expect
+
+describe.only("Rewards/Upgrades", () => {
+    const owner = accounts[0]
+    const thirdParty = accounts[1]
+    const beneficiary = accounts[3]
+
+    const intervalWeights = [5, 10, 15, 20]
+    const totalRewards = 1000000
+    const minimumKeepsPerInterval = 1
+    const termLength = 1000
+    let timestamps
+    let firstIntervalStart
+
+    let token, rewards, newRewards
+
+    before(async () => {
+        token = await KeepToken.new({ from: owner })
+        newRewards = await NewRewardsStub.new()
+
+        firstIntervalStart = await time.latest()
+        timestamps = [
+            101, 150,    // 2 keep in interval 0
```

Comment:
Since here we are in beacon land, maybe we should use "groups" instead of "keeps"?

Reply:
`Rewards` contract thinks in terms of "keeps" and `BeaconRewards` contract translates "keeps" into "groups". Since we are testing the abstract `Rewards` contract here, this test talks about "keeps".

Reply:
Okay, it was just a bit confusing to me, because in `keep-core` we mostly use "groups" and this abstract rewards contract is in `keep-core` whereas in `keep-ecdsa` we use "keeps". Anyway, we can leave it.